src/Storages/StoragePostgreSQL.cpp

```diff
@@ -216,6 +221,63 @@ void StoragePostgreSQL::read(
     query_plan.addStep(std::move(reading));
 }
 
+class SerializationPostgreSQLFixedString : public SerializationFixedString
+{
+private:
+    size_t size;
+
+public:
+    explicit SerializationPostgreSQLFixedString(size_t n_) : SerializationFixedString(n_), size(n_) {}
+
+    void serializeText(const IColumn & column, size_t row_num, WriteBuffer & ostr, const FormatSettings &) const override
+    {
+        const auto * nested_column = &column;
+        if (column.isNullable())
+        {
+            const auto * column_nullable = assert_cast<const ColumnNullable *>(&column);
+            nested_column = &column_nullable->getNestedColumn();
+        }
+
+        const auto column_type = nested_column->getDataType();
+        const char * data = nullptr;
+        if (isFixedString(column_type))
+        {
+            data = reinterpret_cast<const char *>(
+                &assert_cast<const ColumnFixedString *>(nested_column)->getChars()[size * row_num]);
+        }
+        else if (isString(column_type))
+            data = assert_cast<const ColumnString &>(column).getDataAt(row_num).data;
+        else
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Can't serialize type {} as a FixedString", nested_column->getName());
+
+        /// Write only the amount of UTF-8 characters actually used. PostgreSQL complains if we pass
```

Comment:
What will happen if the FixedString contains '\0' naturally? For example `'1\03'::FixedString(3)`

Reply:
That's a good point. PostgreSQL's strings do not accept intermediate \0 for strings, so we cannot really do a 1:1 conversion. We have 2 options here: either we throw a failure because of that:

```
pqxx::data_exception: Failure during '[END COPY]': ERROR:  invalid byte sequence for encoding "UTF8": 0x00
```

Or we convert any intermediate `\0` to a whitespace, which is what I did at [`c9bf83a` (#77304)](https://github.com/ClickHouse/ClickHouse/pull/77304/commits/c9bf83af398e403531d60f9fab6f22abbaa4bf8a)